File: rpc/2/monsters.py
#!/usr/bin/env python
while 1:
    try:
        n = int(input())
        lst = [int(x) for x in input().split()]
    except:
        break
    # ind = {x: i for i, x in enumerate(lst)}
    cnt = 0
    for i in range(n):
        while i + 1 != lst[i]:
            idx = lst[i] - 1
            lst[i], lst[idx] = lst[idx], lst[i]
            cnt += 1
    print(cnt)

File: rpc/2/monsters.cpp
#include <stdio.h>

int lst[200001];

int main() {
    int n;
    while (scanf("%d", &n) != EOF) {
        for (int i = 0; i < n; i++) {
            scanf("%d", &lst[i]);
        }
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            while (lst[i] != i + 1) {
                int idx = lst[i] - 1;
                int tmp = lst[i];
                lst[i] = lst[idx];
                lst[idx] = tmp;
                cnt++;
            }
        }
        printf("%d\n", cnt);
    }
}